exchanges/kraken.py

```diff
@@ -9,6 +15,20 @@ def __init__(self):
         self.timeout = 5.0
         self.url = 'https://api.kraken.com/0/'
 
+        # Load the API keys from the data folder.
+        key_file = os.path.join(PRIVATE_DATA_DIR, 'kraken-key')
+        if os.path.exists(key_file):
+            with open(key_file, 'rt') as f:
+                self.public_key = f.readline().rstrip()
+                self.private_key = f.readline().rstrip()
+        else:
+            raise Warning('The Kraken API requires an API key to access '
```

Comment:
The warning does not stop execution, so for those cases you will see a warning and choose to ignore it. You can also disable warnings on a case by case basis. I think by far the more common usage will require the private queries and having a warning will save us headaches.